src/network-validation/network-validation.service.ts

```diff
@@ -0,0 +1,84 @@
+import { MikroORM, UseRequestContext } from '@mikro-orm/core';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { LidoLocator, LIDO_LOCATOR_CONTRACT_TOKEN } from '@lido-nestjs/contracts';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { ConfigService } from '../common/config';
+import { ConsensusProviderService } from '../common/consensus-provider';
+import { ExecutionProviderService } from '../common/execution-provider';
+import { RegistryKeyStorageService, RegistryOperatorStorageService } from '../common/registry';
+import { SRModuleStorageService } from '../storage/sr-module.storage';
+import { AppInfoStorageService } from '../storage/app-info.storage';
+import { MODULE_ADDRESSES_FOR_CHAINS } from './network-validation.constants';
+
+@Injectable()
+export class NetworkValidationService {
+  constructor(
+    protected readonly orm: MikroORM,
+    @Inject(LIDO_LOCATOR_CONTRACT_TOKEN) protected readonly contract: LidoLocator,
```

Comment:
lets rename contract to locatorContract ?
ALso we have LidoLocatorService , maybe we should get address from there

Reply:
I renamed the `contract` to `locatorContract`.

As far as using the `LidoLocatorService` instead of the `LidoLocator` from "@lido-nestjs/contracts". It has only one `getStakingRouter` method to get the staking router address. And this method internally logs the `contract.address` from `LidoLocator` as the locator address.

I'm not sure, do you think that it is better to use the `getStakingRouter` method to get the locator address instead of the `contract.address`? If yes, why?